### PR TITLE
Hypixel API Improvements

### PR DESCRIPTION
**Pull request by Wyvest**
## Description
This PR improves the Hypixel API in OneConfig.

- adds DevAuth so we can login to Hypixel in dev env
- adds a chat send event (for locraw)
- adds a `doesPlayerExist` method in `ServerPlatform`
- replaces Seraph locraw with the old Hytils locraw
  - Seraph's locraw had a lot of random bugs and Hytils just worked better
  - also adds proper attribution to co-authors
- adds `HypixelKey` annotation
  - this allows `Text` fields to be annotated with `HypixelKey`, which allows OneConfig to:
    - automatically set the field when `/api new` is run in chat
    - sync the field with other fields annotated with `HypixelKey`
    - allow users to set all fields annotated with `HypixelKey` to the same value via the Preferences GUI.
  - All of this can be disabled by the user, and marks the field as non-profile specific.
- fixes button methods not showing
- fixes non-profile exclusion strategy ignoring `Exclude`

## Checklist
- [x] I made a clear description of what was changed
- [x] I stated why these changes were necessary
- [x] I updated documentation or said what needs to be updated
- [x] I made sure these changes are backwards compatible
- [x] This pull request is for one feature/bug fix

### CODE REVIEW

**Review by xtrm-en — APPROVED** [review]
lgtm

File: src/main/java/cc/polyfrost/oneconfig/internal/config/HypixelKeys.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+    public void addOption(BasicOption option) {
+        options.add(option);
+    }
+
+    @Subscribe
+    private void onChatReceived(ChatReceiveEvent event) {
+        if (Preferences.autoSetHypixelKey) {
+            String message = event.getFullyUnformattedMessage();
+            if (message.startsWith("Your new API key is ")) {
```
Comment (nxtdaydelivery) [inline src/main/java/cc/polyfrost/oneconfig/internal/config/HypixelKeys.java:65]: maybe check the message length before this check. Just saves a little bit of useless processing time

**Review by nxtdaydelivery — APPROVED** [review]
looks good to me. doing my comment would be nice.

File: src/main/java/cc/polyfrost/oneconfig/internal/config/HypixelKeys.java (hunk 1 above)
Comment (Wyvest) [reply src/main/java/cc/polyfrost/oneconfig/internal/config/HypixelKeys.java:65]: *why*

**Review by TymanWasTaken — APPROVED** [review]

**Review by xtrm-en — APPROVED** [review]
Lgtm!

**Review by pauliesnug — CHANGES_REQUESTED** [review]
everything else lgtm

File: src/main/java/cc/polyfrost/oneconfig/config/Config.java [hunk 2]
```diff
@@ -193,17 +213,18 @@ protected final void generateOptionList(Object instance, OptionPage page, Mod mo
...
             Button button = ConfigUtils.findAnnotation(method, Button.class);
             String optionName = pagePath + method.getName();
             if (button != null) {
-                ConfigButton option = ConfigButton.create(method, instance);
-                ConfigUtils.getSubCategory(page, button.category(), button.subcategory()).options.add(option);
+                BasicOption option = ConfigUtils.addOptionToPage(page, method, instance);
                 optionNames.put(optionName, option);
             }
-        }*/
+        }
```
Comment (DeDiamondPro) [inline src/main/java/cc/polyfrost/oneconfig/config/Config.java:227]: can't wait for all buttons using methods to just be at the bottom

Comment (Wyvest) [reply src/main/java/cc/polyfrost/oneconfig/config/Config.java:227]: > can't wait for all buttons using methods to just be at the bottom

I fixed it

File: src/main/java/cc/polyfrost/oneconfig/config/Config.java [hunk 3]
```diff
@@ -168,10 +180,18 @@ protected final void generateOptionList(Object instance, OptionPage page, Mod mo
         for (Field field : instance.getClass().getDeclaredFields()) {
             Option option = ConfigUtils.findAnnotation(field, Option.class);
             CustomOption customOption = ConfigUtils.findAnnotation(field, CustomOption.class);
+            boolean isHypixelKey = field.isAnnotationPresent(HypixelKey.class);
```
[the remark this answers is not in the document]
Comment (Wyvest) [reply src/main/java/cc/polyfrost/oneconfig/config/Config.java:183]: We can do that when we do add more special config options

**Review by pauliesnug — APPROVED** [review]
i'm not happy about this

**Review by xtrm-en — APPROVED** [review]
rereview

**Review by pauliesnug — DISMISSED** [review]
rerrereview

**Review by pauliesnug — DISMISSED** [review]
looks gay to me

**Review by xtrm-en — APPROVED** [review]
lgtm

**Review by CaledonianEH — APPROVED** [review]
Ready to be pushed

**Review by xtrm-en — APPROVED** [review]
looks gay to me

**Pull request merged**